### PR TITLE
Add multi module maven support

### PR DESCRIPTION
**Pull request by beyang**
@sqs This branch is ahead of master and its tests pass on Travis. Should this be merged?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,6 +18,11 @@ RUN hg clone http://hg.openjdk.java.net/jdk8u/jdk8u /tmp/jdk8u-build
 WORKDIR /tmp/jdk8u-build
 # Really get jdk8u (otherwise get_source.sh gets jdk8 not jdk8u subrepos)
 RUN bash ./get_source.sh
+
+# Apply a patch to fix an issue in the javac API
+ADD ./hg-27bb4-javac-jdk8u-langtools.patch /tmp/hg-27bb4-javac-jdk8u-langtools.patch
+RUN cd langtools && hg pull -r 27bb4c63fd70 && hg update -r 27bb4c63fd70 && patch -p1 < /tmp/hg-27bb4-javac-jdk8u-langtools.patch
```
Comment (beyang) [inline Dockerfile]: Is there any proposal to submit this patch upstream at all?

File: mvn-install-nodeps-errortolerant.bash [hunk 2]
```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
```
Comment (beyang) [inline mvn-install-nodeps-errortolerant.bash]: unused file?

**Comment by sqs** [comment]
Yeah it should be merged - there may be some cleanup necessary though. I think it should not introduce regressions to single module Maven or Gradle but I am not sure (if tests pass, that's a good signal that it does not).

File: src/main/java/com/sourcegraph/javagraph/Resolver.java [hunk 3]
```diff
@@ -63,6 +65,7 @@ public ResolvedTarget resolveOrigin(URI origin) throws Exception {
             resolvedOrigins.put(origin, null);
             return null;
         }
+        // System.err.println("## Resolved " + rawDep.toString() + " to " + res.toString());
```
Comment (beyang) [inline src/main/java/com/sourcegraph/javagraph/Resolver.java]: remove

**Pull request merged**